patches/server/0887-Fire-CauldronLevelChange-on-initial-fill.patch

```diff
@@ -0,0 +1,68 @@
++                // Paper start - call event for initial fill
++                if (!LayeredCauldronBlock.changeLevel(state, world, pos, Blocks.POWDER_SNOW_CAULDRON.defaultBlockState(), null, CauldronLevelChangeEvent.ChangeReason.NATURAL_FILL)) {
++                    return;
++                }
++                // Paper end
+                 world.gameEvent((Entity) null, GameEvent.FLUID_PLACE, pos);
+             }
+ 
+@@ -54,11 +62,19 @@ public class CauldronBlock extends AbstractCauldronBlock {
+     @Override
+     protected void receiveStalactiteDrip(BlockState state, Level world, BlockPos pos, Fluid fluid) {
+         if (fluid == Fluids.WATER) {
+-            LayeredCauldronBlock.changeLevel(state, world, pos, Blocks.WATER_CAULDRON.defaultBlockState(), null, CauldronLevelChangeEvent.ChangeReason.NATURAL_FILL); // CraftBukkit
++            // Paper start - don't send level event or game event if cancelled
++            if (!LayeredCauldronBlock.changeLevel(state, world, pos, Blocks.WATER_CAULDRON.defaultBlockState(), null, CauldronLevelChangeEvent.ChangeReason.NATURAL_FILL)) { // CraftBukkit
++                return;
++            }
++            // Paper end
+             world.levelEvent(1047, pos, 0);
+             world.gameEvent((Entity) null, GameEvent.FLUID_PLACE, pos);
+         } else if (fluid == Fluids.LAVA) {
+-            LayeredCauldronBlock.changeLevel(state, world, pos, Blocks.LAVA_CAULDRON.defaultBlockState(), null, CauldronLevelChangeEvent.ChangeReason.NATURAL_FILL); // CraftBukkit
++            // Paper start - don't send level event or game event if cancelled
++            if (!LayeredCauldronBlock.changeLevel(state, world, pos, Blocks.LAVA_CAULDRON.defaultBlockState(), null, CauldronLevelChangeEvent.ChangeReason.NATURAL_FILL)) { // CraftBukkit
++                return;
++            }
++            // Paper end
+             world.levelEvent(1046, pos, 0);
+             world.gameEvent((Entity) null, GameEvent.FLUID_PLACE, pos);
+         }
+diff --git a/src/main/java/net/minecraft/world/level/block/LayeredCauldronBlock.java b/src/main/java/net/minecraft/world/level/block/LayeredCauldronBlock.java
+index e6ea389350cf391a87c4c388ed9a6325bdceb90d..c21b0e7265488f26179810ddb6a8a6992a2a4807 100644
+--- a/src/main/java/net/minecraft/world/level/block/LayeredCauldronBlock.java
++++ b/src/main/java/net/minecraft/world/level/block/LayeredCauldronBlock.java
+@@ -89,7 +89,7 @@ public class LayeredCauldronBlock extends AbstractCauldronBlock {
+     }
+ 
+     // CraftBukkit start
+-    public static boolean changeLevel(BlockState iblockdata, Level world, BlockPos blockposition, BlockState newBlock, Entity entity, CauldronLevelChangeEvent.ChangeReason reason) {
++    public static boolean changeLevel(BlockState iblockdata, Level world, BlockPos blockposition, BlockState newBlock, @javax.annotation.Nullable Entity entity, CauldronLevelChangeEvent.ChangeReason reason) { // Paper - entity is nullable
```

Comment:
adding extra diffs to add nullable annotations to Vanilla (or CB extended Vanilla) methods is meh

Reply:
I guess machine added it because IJ was screaming, and I kinda agree with trying to make IJ inspections reliable again. after hard fork this will have no maintiance impact